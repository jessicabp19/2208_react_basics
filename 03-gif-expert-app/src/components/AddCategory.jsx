import { useState } from 'react'

 
 export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) => {
        // console.log(target.value);
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault(); //Evita que se recargue la p√°gina
        const cleanInput = inputValue.trim()
        if (cleanInput.length <= 1 ) return;
        // setCategories( categories => [inputValue, ...categories] );
        onNewCategory(cleanInput)
        setInputValue("");
    }

   return (

    <form onSubmit={ onSubmit }> 
    {/* (event) => onSubmit(event) */}
        <input 
            type="text"
            placeholder="Buscar Gifs"
            value={ inputValue }
            onChange={ onInputChange }
        />
    </form>

   )
 }
 