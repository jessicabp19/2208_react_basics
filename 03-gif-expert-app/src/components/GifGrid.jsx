import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

  /* Custom Hook - { } */
  const { images, isLoading } = useFetchGifs( category );

  return (

    
    <>
        <h3> {category} </h3>
        {/* <LoadingMessage isLoading={isLoading} /> */}
        {
          // isLoading ? <h2> Cargando ... </h2> : null
          isLoading && <h2> Cargando ... </h2>
        }

        <div className="card-grid">
          { 
            images.map( (image) => ( //{id, title}
              <GifItem 
                key={image.id}
                { ...image } //Operador spread
              />
            ))
          }
        </div>
    </>
  )
}
