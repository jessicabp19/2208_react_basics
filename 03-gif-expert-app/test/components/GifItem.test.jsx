import { GifItem } from "../../src/components/GifItem";
import { render, screen } from "@testing-library/react";

describe('Tests in <GifItem />', () => {

    const title = 'Grey\'s Anatomy';
    const url   = 'https://greysanatomy/serie';

    test('Should match snapshot', () => {
        
        const { container } = render( <GifItem title={title} url={url} /> );
        expect( container ).toMatchSnapshot();

    });

    test('Should show the correct image and url', () => {

        render( <GifItem title={title} url={url} /> );
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
        
    });

    test('Should verify the title', () => {

        render( <GifItem title={title} url={url} /> );
        expect( screen.getByText( title ) ).toBeTruthy();

    });

});