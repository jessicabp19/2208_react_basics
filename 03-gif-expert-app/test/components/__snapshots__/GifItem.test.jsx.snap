// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifItem /> Should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Grey's Anatomy"
      src="https://greysanatomy/serie"
    />
    <p>
       
      Grey's Anatomy
       
    </p>
  </div>
</div>
`;
